### ISSUE TITLE
dnf5 --installroot /the/root doesn't read  /the/root/etc/dnf/dnf.conf

### ISSUE THREAD

**Issue by praiskup** [rpm-software-management/dnf5#248]
From `man 7 dnf5-installroot`

       ...
       configuration  file  and  reposdir  are searched inside the installroot
       first.  If they are not present, they are taken from the host system.

       Note: When a path is specified within a command line  argument  (--con‐
       fig=CONFIG_FILE_PATH  in case of configuration file and --setopt=repos‐
       dir=REPO_DIR_PATH for reposdir), then this path is always  relative  to
       the host with no exceptions.
       ...

The first paragraph is wrong, but the second is correct;  using a host-relative
path to the correct dnf.conf works-around the problem.

edit: we should fix dnf5 so it reads the config from --installroot, if present

**Comment by praiskup** [comment]
Blocks: https://github.com/rpm-software-management/mock/pull/1027

**Comment by praiskup** [comment]
Just to make it clear;  IMO it is definitely desired that DNF5 reads the configuration
from the --installroot directory (if present).  I am not requesting a documentation fix.

**Comment by j-mracek** [comment]
The issue will be resolved as a part of bigger task. Do you need to resolve your issue first ar can you wait until we will heve the complete solution related usage of installroot.

**Comment by praiskup** [comment]
It would be nice to have this fixed rather sooner than later.  I think we
should have Mock prepared before Fedora switches to DNF5.

I'm not sure what is the ETA for the bigger --installroot task, and how
quickly is this issue in particular fixable.  So intuitively I bet it is
the DNF team who has to decide.  Not sure.

The thing is that having this fixed, I could finish the fixes on Mock side
(related to DNF5 support) and release.  Then DNF team can start
experimenting with mock, etc.  The idea was to prepare an early CI task
for Mock+DNF5, so we don't bring some accidental breakage during the GA
time.

**Comment by praiskup** [comment]
Since #271 is merged, I think we can close this.  Do we have a testing build somewhere?

**Comment by m-blaha** [comment]
We have nightly builds in https://copr.fedorainfracloud.org/coprs/rpmsoftwaremanagement/dnf5-unstable/ copr repository.

**Comment by praiskup** [comment]
Did you want to write https://copr.fedorainfracloud.org/coprs/rpmsoftwaremanagement/dnf-nightly/ ?

**Comment by j-mracek** [comment]
Yoy can try to download builds from PR checks - see https://github.com/rpm-software-management/dnf5/pull/271/checks

**Comment by m-blaha** [comment]
We probably should rename the repo to something less scary - like `dnf5-nightly`.

**Comment by j-mracek** [comment]
> We probably should rename the repo to something less scary - like `dnf5-nightly`.

Good sugestion

**Comment by praiskup** [comment]
Ok, I have no rights to reopen this issue - but there still is some issue.  Tested with `dnf5`
from  [this PR directory](https://copr.fedorainfracloud.org/coprs/rpmsoftwaremanagement/CI-dnf5-pr271/):
```
/usr/bin/dnf5 --installroot /var/lib/mock/fedora-rawhide-x86_64-bootstrap/root/ --releasever 39 install -y dnf5 dnf5-plugins --setopt=tsflags=nocontexts
Updating and loading repositories:
Repositories loaded.
No repositories were loaded from the installroot. To use the configuration and repositories of the host system, pass --use-host-config.
Failed to resolve the transaction:
No match for argument: dnf5
No match for argument: dnf5-plugins
```

Note the configuration _exists_ in the chroot:
```
$ find /var/lib/mock/fedora-rawhide-x86_64-bootstrap/root/ | grep -e dnf -e yum
/var/lib/mock/fedora-rawhide-x86_64-bootstrap/root/var/lib/yum
/var/lib/mock/fedora-rawhide-x86_64-bootstrap/root/var/log/yum.log
/var/lib/mock/fedora-rawhide-x86_64-bootstrap/root/var/log/dnf5.log
/var/lib/mock/fedora-rawhide-x86_64-bootstrap/root/var/cache/dnf
/var/lib/mock/fedora-rawhide-x86_64-bootstrap/root/var/cache/yum
/var/lib/mock/fedora-rawhide-x86_64-bootstrap/root/etc/dnf
/var/lib/mock/fedora-rawhide-x86_64-bootstrap/root/etc/dnf/vars
/var/lib/mock/fedora-rawhide-x86_64-bootstrap/root/etc/dnf/dnf.conf
/var/lib/mock/fedora-rawhide-x86_64-bootstrap/root/etc/yum.conf
/var/lib/mock/fedora-rawhide-x86_64-bootstrap/root/etc/yum
/var/lib/mock/fedora-rawhide-x86_64-bootstrap/root/etc/yum/yum.conf
```

**Comment by praiskup** [comment]
```
dnf5-5.0.7-20230313200530.19.gd8542f2f.fc37.x86_64
librepo-1.15.1-1.fc37.x86_64
```

**Comment by evan-goode** [comment]
Oops, thanks @praiskup, loading repos from the installroot config file should be fixed by https://github.com/rpm-software-management/dnf5/pull/394.
